### PR TITLE
Cleaning up maven properties

### PR DESCRIPTION
**Pull request by joakime**
+ Ensuring all dependencies and plugins have properties
+ Organized in top level pom (by purpose and then alphabetized)
+ All dependency properties end in ".version"
+ All plugin properties end in ".plugin.version"

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,66 +12,116 @@
...
+    <versions.plugin.version>2.8.1</versions.plugin.version>
 
     <!-- testing -->
+    <awaitility.version>4.1.0</awaitility.version>
     <it.debug>false</it.debug>
     <jetty.test.version>5.8</jetty.test.version>
-    <spring-boot.version>2.1.1.RELEASE</spring-boot.version>
+    <jetty.testtracker.log>false</jetty.testtracker.log>
     <jsr250-api.version>1.0</jsr250-api.version>
-    <javax.annotation-api>1.3.2</javax.annotation-api>
```
Comment (olamy) [inline pom.xml:69]: this need to updated in some maven plugin its as well

**Review by olamy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (joakime) [reply pom.xml:69]: It's there in line 44 of the main pom.xml

https://github.com/eclipse/jetty.project/blob/505874110edcd040aebd3c52c9de9f0ec03112b0/pom.xml#L44

**Comment by olamy** [comment]
Yes but annotation.api and not annotation-api ;) which I guess the ITs are
using

**Comment by joakime** [comment]
I'm fixing the `jetty-maven-plugin` it tests now.

**Review by olamy — APPROVED** [review]

**Pull request merged**